src/Mod/Part/App/AttachExtension.cpp

```diff
@@ -477,6 +482,164 @@ bool AttachExtension::extensionHandleChangedPropertyName(Base::XMLReader& reader
     return App::DocumentObjectExtension::extensionHandleChangedPropertyName(reader, TypeName, PropName);
 }
 
+void AttachExtension::handleLegacyTangentPlaneOrientation()
+{
+    // check attachment mode
+    if (_props.attacher->mapMode != mmTangentPlane) {
+        return;
+    }
+
+    // check stored document program version (applies to v1.0 and earlier only)
+    int major, minor;
+    if (sscanf(getExtendedObject()->getDocument()->getProgramVersion(), "%d.%d", &major, &minor) != 2) {
+        return;
+    }
+    if (major > 1 || (major == 1 && minor > 0)) {
+        return;
+    }
+
+    // check for an App::Plane support object exists
+    App::GeoFeature* geof = nullptr;
+    for (auto obj : this->AttachmentSupport.linkedObjects()) {
+        if (obj->isDerivedFrom(Base::Type::fromName("App::Plane"))) {
+            geof = dynamic_cast<App::GeoFeature*>(obj);
```

Comment:
You can make use of the FreeCAD type system here:

```suggestion
            geof = freecad_cast<App::GeoFeature*>(obj);
```